be/src/vec/exec/join/process_hash_table_probe_impl.h

```diff
@@ -207,883 +208,601 @@ void ProcessHashTableProbe<JoinOpType>::_pre_serialize_key(
                 } else {
-                    for (int i = last_offset; i < _probe_indexs.size(); ++i) {
-                        _probe_indexs[i] = current_probe_index;
-                    }
-                    _probe_indexs.resize(current_offset, current_probe_index);
-                }
-                all_match_one &= (count == 1);
-                if (current_offset >= _batch_size) {
-                    break;
+                    ++probe_index;
                 }
             }
-            probe_size = probe_index - last_probe_index + probe_row_match_iter.ok();
+            all_match_one &= (current_offset == _probe_indexs.size() + 1);
+            _probe_indexs.resize(current_offset, current_probe_index);
         }
+        probe_size = probe_index - last_probe_index + probe_row_match_iter.ok();
     }
 
-    {
-        SCOPED_TIMER(_build_side_output_timer);
-        build_side_output_column(mcol, right_col_idx, right_col_len,
-                                 *_join_context->_right_output_slot_flags, current_offset);
-    }
+    build_side_output_column(mcol, *_join_context->_right_output_slot_flags, current_offset,
+                             with_other_conjuncts);
 
-    if constexpr (JoinOpType != TJoinOp::RIGHT_SEMI_JOIN &&
-                  JoinOpType != TJoinOp::RIGHT_ANTI_JOIN) {
-        SCOPED_TIMER(_probe_side_output_timer);
+    if constexpr (with_other_conjuncts || (JoinOpType != TJoinOp::RIGHT_SEMI_JOIN &&
+                                           JoinOpType != TJoinOp::RIGHT_ANTI_JOIN)) {
         RETURN_IF_CATCH_EXCEPTION(probe_side_output_column(
                 mcol, *_join_context->_left_output_slot_flags, current_offset, last_probe_index,
-                probe_size, all_match_one, false));
+                probe_size, all_match_one, with_other_conjuncts));
     }
 
     output_block->swap(mutable_block.to_block());
 
+    if constexpr (with_other_conjuncts) {
+        return do_other_join_conjuncts(output_block, is_mark_join, multi_matched_output_row_count,
+                                       is_the_last_sub_block);
+    }
+
     return Status::OK();
 }
 
 template <int JoinOpType>
-template <bool need_null_map_for_probe, bool ignore_null, typename HashTableType>
-Status ProcessHashTableProbe<JoinOpType>::do_process_with_other_join_conjuncts(
-        HashTableType& hash_table_ctx, ConstNullMapPtr null_map, MutableBlock& mutable_block,
-        Block* output_block, size_t probe_rows, bool is_mark_join) {
-    auto& probe_index = *_join_context->_probe_index;
-    auto& probe_raw_ptrs = *_join_context->_probe_columns;
-    if (_build_block_rows.size() < probe_rows * PROBE_SIDE_EXPLODE_RATE) {
-        _probe_indexs.resize(probe_rows * PROBE_SIDE_EXPLODE_RATE);
-        _build_block_rows.resize(probe_rows * PROBE_SIDE_EXPLODE_RATE);
-        _build_block_offsets.resize(probe_rows * PROBE_SIDE_EXPLODE_RATE);
+Status ProcessHashTableProbe<JoinOpType>::do_other_join_conjuncts(
```

Comment:
add comment for the param：
what means of：
1、is_the_last_sub_block
2. multi_matched_output_row_count

Reply:
Comments are added to variable declarations